src/stores/ProfileController.ts

```diff
@@ -1,38 +1,42 @@
-import useService from '#src/hooks/useService';
+import { injectable } from 'inversify';
+
 import type { ProfilePayload, EnterProfilePayload, ProfileDetailsPayload } from '#types/account';
+import { ProfileService } from '#src/services/profile.service';
+import { useConfigStore } from '#src/stores/ConfigStore';
+
+@injectable()
+export default class ProfileController {
+  private readonly profileService: ProfileService;
+
+  constructor(profileService: ProfileService) {
+    this.profileService = profileService;
+  }
+
+  listProfiles = () => {
+    return this.profileService?.listProfiles(undefined, this.getSandbox());
+  };
+
+  createProfile = ({ name, adult, avatar_url, pin }: ProfilePayload) => {
+    return this.profileService?.createProfile({ name, adult, avatar_url, pin }, this.getSandbox());
+  };
+
+  updateProfile = ({ id, name, adult, avatar_url, pin }: ProfilePayload) => {
+    return this.profileService?.updateProfile({ id, name, adult, avatar_url, pin }, this.getSandbox());
+  };
+
+  enterProfile = ({ id, pin }: EnterProfilePayload) => {
+    return this.profileService?.enterProfile({ id, pin }, this.getSandbox());
+  };
+
+  deleteProfile = ({ id }: ProfileDetailsPayload) => {
+    return this.profileService?.deleteProfile({ id }, this.getSandbox());
+  };
+
+  getProfileDetails = ({ id }: ProfileDetailsPayload) => {
+    return this.profileService?.getProfileDetails({ id }, this.getSandbox());
+  };
```

Comment:
Won't this result in the return type being a union `undefined | Promise<Result>`?

Reply:
Yes, that's very much intentional. Downstream code needs to understand that there might not be a profile service / profile controller / profile data. This is true elsewhere for things like checkout too.

Reply:
Ah I understand. So this can replace the feature flags 😄